kernel/src/syscall.rs

```diff
@@ -252,53 +253,53 @@ impl Syscall {
     pub fn from_register_arguments(
         syscall_number: u8,
         r0: usize,
-        r1: usize,
-        r2: usize,
-        r3: usize,
+        r1: CapabilityPtr,
+        r2: CapabilityPtr,
+        r3: CapabilityPtr,
     ) -> Option<Syscall> {
```

Comment:
I think for the same reason we were using `usize` before, when we would sometimes actually pass values such as `u8` inside them (using `as` internally as appropriate). We do not know yet which width each argument will be, so we just choose the largest.

`CapabilityPtr` is guaranteed to be wide enough to pass `usize`, but not the reverse.

Reply:
Hmm, I agree that the `u` in `usize` has the same problem.

If we are going to change this I want to right the wrongs of the past and for this to be a type called `Register` (or similar). I think it behooves us to be able to clarify exactly what we mean, both for the OS implementation, but also for future readers of the code. This argument has no semantic meaning other than the number of bits that fit in a CPU register.

I don't like using `type`, but I would even settle for just `type Register = CapabilityPtr` because then at least we have a place where we can comment to explain why that mapping is valid and correct and why we made the change (and not just use usize). I think if I was trying to understand the code base I would assume that since `usize` is the normally the way to say size of a register, using a pointer type must mean something, probably related to pointers.

I also think we benefit from separating where we want to convey something related to authority and where we just need to store a register's worth of bits. Right now while we have been thinking about this extensively the difference is probably easy to keep track of. But in a year, or for someone not involved in this discussion, that might not be true. Then we risk that we handle it incorrectly at that point.